src/tiled/addpropertydialog.cpp

```diff
@@ -46,24 +48,36 @@ AddPropertyDialog::AddPropertyDialog(QWidget *parent)
     mUi->setupUi(this);
     resize(Utils::dpiScaled(size()));
 
+
     // Add possible types from QVariant
     mUi->typeBox->addItem(typeToName(QVariant::Bool),    false);
     mUi->typeBox->addItem(typeToName(QVariant::Color),   QColor());
     mUi->typeBox->addItem(typeToName(QVariant::Double),  0.0);
     mUi->typeBox->addItem(typeToName(filePathTypeId()),  QVariant::fromValue(FilePath()));
     mUi->typeBox->addItem(typeToName(QVariant::Int),     0);
+    
     mUi->typeBox->addItem(typeToName(objectRefTypeId()), QVariant::fromValue(ObjectRef()));
     mUi->typeBox->addItem(typeToName(QVariant::String),  QString());
 
+
+    CustomProps customProps = Object::customProps();
+
+
+    for (const CustomProp cProp: customProps) {
+        QVariant var;
+        var.setValue(cProp);
+        mUi->typeBox->addItem(cProp.name,  var);
+    }
+        
     mUi->buttonBox->button(QDialogButtonBox::Ok)->setEnabled(false);
 
     // Restore previously used type
     mUi->typeBox->setCurrentText(session::propertyType);
 
     connect(mUi->name, &QLineEdit::textChanged,
-            this, &AddPropertyDialog::nameChanged);
+        this, &AddPropertyDialog::nameChanged);
```

Comment:
Please leave the original alignment intact. The coding style I follow for wrapped parameters aligns them to the opening bracket.